x/finality/keeper/liveness.go

```diff
@@ -74,54 +82,28 @@ func (k Keeper) HandleFinalityProviderLiveness(ctx context.Context, fpPk *types.
 	minHeight := signInfo.StartHeight + signedBlocksWindow
 	maxMissed := signedBlocksWindow - minSignedPerWindow
 
-	// if we are past the minimum height and the finality provider has missed too many blocks, punish them
+	// if the number of missed block reaches the threshold within the sliding window
+	// jail the finality provider
 	if height > minHeight && signInfo.MissedBlocksCounter > maxMissed {
 		updated = true
 
-		k.Logger(sdkCtx).Info(
-			"detected sluggish finality provider",
-			"height", height,
-			"public_key", fpPk.MarshalHex(),
-			"missed_count", signInfo.MissedBlocksCounter,
-			"threshold", minSignedPerWindow,
-			"window_size", signedBlocksWindow,
-		)
-
-		// Inactivity detected
-		err = k.hooks.AfterSluggishFinalityProviderDetected(ctx, fpPk)
-		if err != nil {
-			return err
+		if err := k.jailSluggishFinalityProvider(ctx, fpPk); err != nil {
+			return fmt.Errorf("failed to jail sluggish finality provider %s: %w", fpPk.MarshalHex(), err)
 		}
 
-		if err := sdkCtx.EventManager().EmitTypedEvent(
-			finalitytypes.NewEventSluggishFinalityProviderDetected(fpPk),
-		); err != nil {
-			panic(fmt.Errorf("failed to emit sluggish finality provider detected event for height %d: %w", height, err))
+		signInfo.JailedUntil = sdkCtx.HeaderInfo().Time.Add(params.JailDuration)
```

Comment:
Just to confirm, this value will be checked in `MsgUnjail` in subsequent PRs right?

Reply:
Exactly